### PR TITLE
Structure stuff, based on liach's PR

### PR DESCRIPTION
**Pull request by apple502j**
Supersedes #2729

### CODE REVIEW

File: mappings/net/minecraft/structure/Structure.mapping [hunk 1]
```diff
@@ -115,6 +121,40 @@ CLASS net/minecraft/class_3499 net/minecraft/structure/Structure
...
+		COMMENT Categorizes {@code blockInfo} based on its properties, modifying
+		COMMENT the passed lists in-place.
+		COMMENT
+		COMMENT <p>If the block has an NBT associated with it, then it will be
+		COMMENT put in {@code blocksWithNbt}. If the block does not have an NBT
+		COMMENT associated with it, but is always a full cube, then it will be
+		COMMENT put in {@code fullBlocks}. Otherwise, it will be put in
+		COMMENT {@code otherBlocks}.
+		COMMENT
+		COMMENT @apiNote After all blocks are categorized, {@link combineSorted}
```
Comment (enbrain) [inline mappings/net/minecraft/structure/Structure.mapping]:
```suggestion
		COMMENT @apiNote After all blocks are categorized, {@link #combineSorted}
```

File: mappings/net/minecraft/structure/Structure.mapping [hunk 2]
```diff
@@ -115,6 +121,40 @@ CLASS net/minecraft/class_3499 net/minecraft/structure/Structure
...
+		COMMENT shoud be called with the same parameters to get the final list.
+		ARG 0 blockInfo
+		ARG 1 fullBlocks
+		ARG 2 blocksWithNbt
+		ARG 3 otherBlocks
+	METHOD method_28055 combineSorted (Ljava/util/List;Ljava/util/List;Ljava/util/List;)Ljava/util/List;
+		COMMENT {@return the list that sorts and combines the passed block lists}
+		COMMENT
+		COMMENT @apiNote The parameters passed shoule be the same one that was passed
+		COMMENT to previous calls to {@link categorize}. The returned value is meant to
```
Comment (enbrain) [inline mappings/net/minecraft/structure/Structure.mapping]:
```suggestion
		COMMENT to previous calls to {@link #categorize}. The returned value is meant to
```

File: mappings/net/minecraft/structure/Structure.mapping [hunk 3]
```diff
@@ -115,6 +121,40 @@ CLASS net/minecraft/class_3499 net/minecraft/structure/Structure
...
+		COMMENT be passed to {@link PalettedBlockInfoList}.
```
Comment (enbrain) [inline mappings/net/minecraft/structure/Structure.mapping:147]:
```suggestion
		COMMENT be passed to {@link #PalettedBlockInfoList}.
```

File: mappings/net/minecraft/structure/Structure.mapping [hunk 4]
```diff
@@ -115,6 +121,40 @@ CLASS net/minecraft/class_3499 net/minecraft/structure/Structure
...
+		COMMENT the passed lists in-place.
+		COMMENT
+		COMMENT <p>If the block has an NBT associated with it, then it will be
+		COMMENT put in {@code blocksWithNbt}. If the block does not have an NBT
+		COMMENT associated with it, but is always a full cube, then it will be
+		COMMENT put in {@code fullBlocks}. Otherwise, it will be put in
+		COMMENT {@code otherBlocks}.
+		COMMENT
+		COMMENT @apiNote After all blocks are categorized, {@link combineSorted}
+		COMMENT shoud be called with the same parameters to get the final list.
```
Comment (enbrain) [inline mappings/net/minecraft/structure/Structure.mapping]:
```suggestion
		COMMENT should be called with the same parameters to get the final list.
```

File: mappings/net/minecraft/structure/Structure.mapping [hunk 5]
```diff
@@ -115,6 +121,40 @@ CLASS net/minecraft/class_3499 net/minecraft/structure/Structure
...
+		ARG 0 blockInfo
+		ARG 1 fullBlocks
+		ARG 2 blocksWithNbt
+		ARG 3 otherBlocks
+	METHOD method_28055 combineSorted (Ljava/util/List;Ljava/util/List;Ljava/util/List;)Ljava/util/List;
+		COMMENT {@return the list that sorts and combines the passed block lists}
+		COMMENT
+		COMMENT @apiNote The parameters passed shoule be the same one that was passed
```
Comment (enbrain) [inline mappings/net/minecraft/structure/Structure.mapping]:
```suggestion
		COMMENT @apiNote The parameters passed should be the same one that was passed
```

File: mappings/net/minecraft/structure/Structure.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/structure/Structure.mapping:147]: hmm, don't think you use hash for nested or inner classes

**Review by enbrain — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Review by liach — APPROVED** [review]
Looks good in a glance from web review

**Pull request merged**